TF-3779 add _, isocompoent, .eslintrc, *.stories, and *.config to regex

joecan opened the pull request:
##### *To:*
@kevinzang @udemy/team-f @inancsevinc

##### *What:*
This adds adds:
   - Allow a leading underscore in the filename, e.g. _my-private-
   - Allow .react-isocomponent.js, .react-isocomponent.spec.js
   - Allow .eslintrc.js, *.config.js, *.stories.js

I also broke up the already long regex into joins for commenting.

##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3779

##### *What did you test:*
Added changes to static/src/.eslintrc.js and removed disables in *.react-isocomponent.js, .eslintrc.js, *.config.js and ran:
TIMING=1 yarn lint
I also created some misnamed files and they seemed to be caught.

##### *What dashboards will you be monitoring:*
N/A

joecan committed:
add _, isocompoent, .eslintrc, and *.config to regex

joecan committed:
add *.stories.js to match-regex

kevinzang commented on packages/eslint-config-udemy-website/index.js:
```diff
@@ -23,7 +23,22 @@ module.exports = {
+                    '^(?:_?[a-z0-9\\-]+', // allows underscore start and any lowercase filename
+                    '(?:\\.(?:', // any dot must be:
+                    'ng-(?:constant|controller|directive|factory|filter|provider|service)', // an Angular file
+                    '|videojs-component', // or a videojs-compenent
```
compenent -> component

kevinzang commented on packages/eslint-config-udemy-website/index.js:
```diff
@@ -23,7 +23,22 @@ module.exports = {
+                    'ng-(?:constant|controller|directive|factory|filter|provider|service)', // an Angular file
+                    '|videojs-component', // or a videojs-compenent
+                    '|react-(?:isocomponent|component|proptypes)', // or a React file
+                    '|mobx-(?:model|store)))?', // or a Mobx file
```
Maybe move the last `))?` to a new line to match with `'(?:\\.(?:', // any dot must be:` on its own line.

joecan replied:
I'll do that. We could also get really fancy and make all of the groupings their own array. ;-)

kevinzang reviewed (approved):
@joecan LGTM!

joecan committed:
fix typo in comment and breakout line closing dot group

joecan committed:
Publish

 - babel-polyfill-udemy-website@9.0.4
 - babel-preset-udemy-website@11.0.1
 - eslint-config-udemy-website@12.0.3
